routers/api/v1/repo/pull.go

```diff
@@ -1110,9 +1110,18 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 	// Check if current user has fork of repository or in the same repository.
 	headRepo := repo_model.GetForkedRepo(ctx, headUser.ID, baseRepo.ID)
 	if headRepo == nil && !isSameRepo {
-		log.Trace("parseCompareInfo[%d]: does not have fork or in same repository", baseRepo.ID)
-		ctx.NotFound("GetForkedRepo")
-		return nil, nil, nil, "", ""
+		// Check if the base repository is a fork of the head repository.
+		headRepo, err = repo_model.GetRepositoryByID(ctx, baseRepo.ForkID)
```

Comment:
If that, baseRepo should be headRepo and headRepo should be baseRepo.

Reply:
Ah indeed, the check should be as the comment describes. Working on a revision right now.

Reply:
I'm not sure what the policy is on resolving comments, so I'll leave this open for now commit 1c94e6d should address the comments you made :slightly\_smiling\_face: